js/app/dashboard/redux/taskListEntityReducers.js

```diff
@@ -0,0 +1,106 @@
+import {
+  MODIFY_TASK_LIST_REQUEST,
+  MODIFY_TASK_LIST_REQUEST_SUCCESS,
+  TASK_LIST_UPDATED,
+  UPDATE_TASK
+} from "./actions";
+import {
+  taskUtils,
+  taskListUtils,
+  taskListEntityUtils,
+} from '../../coopcycle-frontend-js/logistics/redux'
+
+const initialState = {
+  byId: {}
+}
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case MODIFY_TASK_LIST_REQUEST: {
+      let entity = taskListEntityUtils.findTaskListByUsername(state.byId, action.username)
+
+      if (entity === undefined) {
+        // eslint-disable-next-line no-console
+        console.assert(false, `entity is undefined: username: ${action.username}`)
+
+        return state
+      }
+
+      let newEntity = {
+        ...entity,
+        itemIds: taskUtils.tasksToIds(action.tasks),
+      }
+
+      let newItems = taskListEntityUtils.addOrReplaceTaskList(state.byId, newEntity)
+
+      return {
+        ...state,
+        byId: newItems,
+      }
+    }
+    case MODIFY_TASK_LIST_REQUEST_SUCCESS: {
+      let newEntity = taskListUtils.replaceTasksWithIds(action.taskList)
+      let newItems = taskListEntityUtils.addOrReplaceTaskList(state.byId, newEntity)
+
+      return {
+        ...state,
+        byId: newItems,
+      }
+    }
+    case TASK_LIST_UPDATED: {
+      let entityByUsername = taskListEntityUtils.findTaskListByUsername(state.byId, action.taskList['username'])
```

Comment:
Problem here. In this case, the task list should be matched by `@id` (not `username`). 

Previous code was like this: 
```es6
taskListIndex = _.findIndex(state.taskLists, taskList => taskList['@id'] === action.taskList['@id'])
```

It should be matched by `@id`, because the `username` of the event may have been generated from a task list for another day. 

It is actually very tricky... each time an address is updated, an async job checks if the address belongs to a task list, and if yes, it recalculates the route for the task list. What can happen, is that someone modifies an address, that belong to a task list from a few days ago. If the same messenger is assigned today, let's say, if we match by username, we get completely wrong data.